psi4/driver/procrouting/scf_proc/scf_iterator.py

```diff
@@ -71,8 +71,8 @@ def scf_compute_energy(self):
             self.initialize()
             try:
                 self.iterations()
-            except ConvergenceError:
-                raise ConvergenceError("""SCF DF preiterations""", self.iteration_)
+            except SCFConvergenceError:
+                raise SCFConvergenceError("""SCF DF preiterations""", self.iteration_, self)
```

Comment:
Looking through the code we probably need to call `scf_finalize_energy` here. There are many variables set in the `scf_finalize_energy/scf_print_energies` functions which we should probably change at some point (or just rename the functions).